go/tools/gopackagesdriver/bazel_json_builder.go

```diff
@@ -119,11 +119,13 @@ func (b *BazelJSONBuilder) Build(ctx context.Context, mode LoadMode) ([]string,
 		return nil, fmt.Errorf("found no labels matching the requests")
 	}
 
+	aspects := append(additionalAspects, rulesGoAspect)
```

Comment:
AFAIK the order may be quite significant here. 

@linzhp Is this the correct one (and if so, why?)?

Reply:
This was a syntax simplification assuming the order didn't matter, if it does I'd be happy to change it so rules_go is first.

Reply:
I'm not saying this order is wrong, I simply don't know the aspect well enough to decide this. If I understood the interaction correctly, the latter aspect would see providers added by the former, but maybe that's just not how it works. :-)

Reply:
If the additional aspects need some information from each other, or from the default aspect, they can specify `required_providers` in their aspect definition, then Bazel would run those aspect in the order to satisfy the requirement.

@JamyDev did you test the these aspects don't override each other when that have the same output group and provider?

Reply:
There is possibility for overlap if you output to the same `.pkg.json` file as the default aspect. The packagesdriver [only considers the .pkg.json files](https://github.com/bazelbuild/rules_go/blob/master/go/tools/gopackagesdriver/bazel_json_builder.go#L136-L140), so long as there's no overlap in these files it's fine.  In our (Uber's) case that's not a problem, we output to `ruleName.pkg.json` (just like the default aspect does) and only output anything when we encounter our custom rule. 

Now if the order in the argument here is actually followed this means that if I write a bad custom aspect, it'll get overridden by the rulesGo aspect, unless I directly declare it as a `required_provider`.